pkgs/by-name/au/ausaxs/package.nix

```diff
@@ -0,0 +1,148 @@
+  };
+
+  backward-cpp = fetchFromGitHub {
+    owner = "bombela";
+    repo = "backward-cpp";
+    rev = "0bfd0a07a61551413ccd2ab9a9099af3bad40681";
+    hash = "sha256-nLH8jfdgzmlUTg6zwY/h0HVnDMeC9rvmX1x4Ithu9dI=";
+  };
+
+  catch2 = fetchFromGitHub {
+    owner = "catchorg";
+    repo = "Catch2";
+    rev = "914aeecfe23b1e16af6ea675a4fb5dbd5a5b8d0a";
+    hash = "sha256-2gK+CUpml6AaHcwNoq0tHLr2NwqtMPx+jP80/LLFFr4=";
+  };
+
+  elements = fetchFromGitHub {
+    owner = "cycfi";
+    repo = "elements";
+    rev = "71ecd1f4ebc76967c6812b1872db639784e40a2d";
+    hash = "sha256-F3Dv+QboXfOSaXpbdOeWPtOC8orWGZc8ZBFho/X8Ky8=";
+  };
+
+  nfd = fetchFromGitHub {
+    owner = "btzy";
+    repo = "nativefiledialog-extended";
+    tag = "v1.2.1";
+    hash = "sha256-GwT42lMZAAKSJpUJE6MYOpSLKUD5o9nSe9lcsoeXgJY=";
+  };
+
+  asio = fetchFromGitHub {
+    owner = "chriskohlhoff";
+    repo = "asio";
+    tag = "asio-1-29-0";
+    hash = "sha256-5WSrMe9n+8i/ZyvCsa4MMBguYbSz+7FwH0Z5JfHtRGM=";
+  };
+
+  cycfi_infra = fetchFromGitHub {
+    owner = "cycfi";
+    repo = "infra";
+    rev = "2dff97a4b107eced78e426152f5001a2331cb1cf";
+    hash = "sha256-NmoPYhfsrC5oWFjJ9Ol83sR8aIkyQr6UpaCeZpW58PI=";
+    fetchSubmodules = true;
+  };
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "ausaxs";
+  version = "1.1.2";
+
+  src = fetchFromGitHub {
+    owner = "AUSAXS";
+    repo = "AUSAXS";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-83ZgplSZmY/2DZXG1t+/4+gzlv/YusRFeDRqNuie0JA=";
+  };
+
+  patches = [ ./cmake-no-fetchcontent.patch ];
+
+  postPatch = ''
+    cp --recursive --no-preserve=mode ${dlib} dlib
+    cp --recursive --no-preserve=mode ${thread-pool} thread_pool
+    cp --recursive --no-preserve=mode ${gcem} gcem
+    cp --recursive --no-preserve=mode ${backward-cpp} backward
+    cp --recursive --no-preserve=mode ${catch2} catch2
+    cp --recursive --no-preserve=mode ${nfd} nfd
+    cp --recursive --no-preserve=mode ${elements} elements
+    cp --recursive --no-preserve=mode ${asio} asio
+    cp --recursive --no-preserve=mode ${cycfi_infra} cycfi_infra
+    substituteInPlace executable/CMakeLists.txt \
+      --replace-fail "FetchContent_MakeAvailable(CLI11)" "find_package(CLI11 CONFIG REQUIRED)"
+    substituteInPlace include/crystal/crystal/miller/MillerGenerationFactory.h \
+      --replace-fail "namespace ausaxs::settings::crystal {enum class MillerGenerationChoice;}" '#include "../core/settings/CrystalSettings.h"'
+    patch -p1 -d elements < ${./elements-cmake-no-fetchcontent.patch}
+    substituteInPlace CMakeLists.txt \
+      --replace-fail "-mavx" "${
+        if (stdenv.hostPlatform.isx86_64 && stdenv.hostPlatform.isLinux) then "-msse3" else ""
+      }"
+    sed -i '/#include <iostream>/a\
+    #include <numbers>' source/crystal/miller/FibonacciMillers.cpp
+  '';
+
+  nativeBuildInputs = [
+    cmake
+    doxygen
+    python3
+    pkg-config
+  ];
+
+  buildInputs = [
+    curlMinimal
```

Comment:
I think that curl is just for bootstrapping and fetchers. We should probably just use curl, to also have proper compression support